### PR TITLE
[Darc] Subscription history and retry

### PR DESCRIPTION
**Pull request by mmitche**
Add darc commands for getting the history of a subscription, as well as a retry command to retry specific failed actions.

### CODE REVIEW

**Comment by mmitche** [comment]
@alexperovich This change was motivated by the fact that the current corefx subscription updates are failing.

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/GetSubscriptionHistoryOperation.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                DarcSettings darcSettings = LocalCommands.GetSettings(_options, Logger);
+                // No need to set up a git type or PAT here.
+                Remote remote = new Remote(darcSettings, Logger);
+
+                // Given the ID, grab the subscription history items.
+                var history = await remote.GetSubscriptionHistoryAsync(_options.SubscriptionId);
+
+                foreach (var historyItem in history)
+                {
+                    Console.WriteLine($"{historyItem.Timestamp.Value.LocalDateTime}: ({(historyItem.Success.Value ? "Success" : "Failure")}) - {historyItem.Action}");
```
Comment (jcagme) [inline src/Microsoft.DotNet.Darc/src/Darc/Operations/GetSubscriptionHistoryOperation.cs]: >Console.WriteLine [](start = 20, length = 17)

Use Logger instead, here and other instances of Console.WriteLone

Comment (mmitche) [reply src/Microsoft.DotNet.Darc/src/Darc/Operations/GetSubscriptionHistoryOperation.cs]: Not for the command line output.  It prefixes every line with the logger information, and is not suitable for normal command line output.

Comment (jcagme) [reply src/Microsoft.DotNet.Darc/src/Darc/Operations/GetSubscriptionHistoryOperation.cs]: Oh, right. You are just outputting the subscription info

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/RetrySubscriptionUpdateOperation.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    {
+        RetrySubscriptionUpdateCommandLineOptions _options;
+        public RetrySubscriptionUpdateOperation(RetrySubscriptionUpdateCommandLineOptions options)
+            : base(options)
+        {
+            _options = options;
+        }
+
+        /// <summary>
+        /// Retrieve information about a particular subscription's history.
```
Comment (jcagme) [inline src/Microsoft.DotNet.Darc/src/Darc/Operations/RetrySubscriptionUpdateOperation.cs]: >Retrieve information about a particular subscription's history [](start = 12, length = 62)

This seems to be a copy/paste from GetSubscriptionHistoryOperation. Please update

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/RetrySubscriptionUpdateOperation.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            {
+                // Not found is fine to ignore.  If we get this, it will be an aggregate exception with an inner API exception
+                // that has a response message code of NotFound.  Return success.
+                Console.WriteLine($"Subscription with id '{_options.SubscriptionId}' or subscription update with id '{_options.UpdateId}' was not found.");
+                return Constants.SuccessCode;
+            }
+            catch (ApiErrorException e) when (e.Response.StatusCode == HttpStatusCode.NotAcceptable)
+            {
+                // Not found is fine to ignore.  If we get this, it will be an aggregate exception with an inner API exception
+                // that has a response message code of NotFound.  Return success.
```
Comment (jcagme) [inline src/Microsoft.DotNet.Darc/src/Darc/Operations/RetrySubscriptionUpdateOperation.cs]: Same comment as in NotFound

**Pull request merged**